### ISSUE TITLE
Ugly output from `:printdef` in packages with "--partial-eval"

### ISSUE THREAD

**Issue by msmorgan** [idris-lang/Idris-dev#4207]
# Steps to Reproduce

Create the following two files:

```ipkg
-- File: mylength.ipkg
package mylength
opts = "--partial-eval"
modules = MyLength
```

```idris
-- File: MyLength.idr
module MyLength

myLength : List a -> Nat
myLength [] = 0
myLength (x :: xs) = 1 + myLength xs
```

Then run the following commands:
```
$ idris --build mylength.ipkg
Type checking .\MyLength.idr

$ idris 00mylength-idx.ibc --nobanner
*00mylength-idx> :module MyLength
*00mylength-idx *MyLength> :printdef myLength
myLength : List a -> Nat
myLength [] = PE_fromInteger_dc75de74 0
myLength (x :: xs) = PE_+_6b31ad5c (PE_fromInteger_dc75de74 1) (myLength xs)
*00mylength-idx *MyLength>
```

# Expected Behavior

The output should be the same as if "--partial-eval" was not used.

# Observed Behavior

See above.

**Comment by david-christiansen** [comment]
I'm not sure whether this is properly considered a bug or not. `:printdef` is intended to give the compiler's view on definitions, rather than the original text, because that's useful for understanding elaboration. But on the other hand, this is pretty scary looking. I wonder if it wouldn't be best to assign more readable representations to the names of PE output, rather than reversing it here.

**Comment by jfdm** [comment]
I agree with @david-christiansen here. Perhaps we should also improve the documentation for `:printdef` to ensure that there are no confusions over its intended use.

**Comment by ahmadsalim** [comment]
I agree with @david-christiansen and @jfdm , that the printing should reflect the internal definition.
However, I think in general that pretty printing of our definitions is verbose and it would be nice if we had a more consistent style all around.
